walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,130 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.collectingAndThen;
+import com.google.common.collect.ImmutableList; 
 
-public final class FindMeetingQuery {
+/**
+ * Returns a Collection of times which are available for optional attendees and mandatory attendees.
+ * If optional attendees cannot be fit in, then only returns the times which mandatory attendees can
+ * attend
+ *
+ * @param events A list of events that may need to be accounted for
+ * @param request The specifications for the meeting to be scheduled
+ * @return A Collection of available times for requested meeting
+ */
+public final class FindMeetingQuery implements Comparator<Event> {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    ImmutableList<Event> eventList = events.stream().sorted(new FindMeetingQuery()).collect(collectingAndThen(toList(), ImmutableList::copyOf));
```

Comment:
Looks like you maybe need to run the formatter again.

Reply:
formatted